webflow-hacks/form-hacks/18/18.ts

```diff
@@ -0,0 +1,27 @@
+// when the DOM is ready
+document.addEventListener('DOMContentLoaded', function () {
+  // make an array of invalid domains
+  const invalidDomains = ['gmail.com', 'yahoo.com', 'hotmail.com', 'competitor.com'];
+  // set the fs-hacks selector
+  const SUBMIT_BUTTON_SELECTOR = '[fs-hacks-element="submit"]';
+  const EMAIL_SELECTOR = '[fs-hacks-element="email"]';
+  // get the dom elements
+  const submitBtn = document.querySelector<HTMLButtonElement>(SUBMIT_BUTTON_SELECTOR);
+  const emailInput = document.querySelector<HTMLInputElement>(EMAIL_SELECTOR);
+  // check if the elements exists
+  if (!submitBtn || !emailInput) return;
+  // on submit button click
+  submitBtn.addEventListener('click', function () {
+    // split email at '@' character to get domain
+    const domainPart = emailInput.value.split('@')[1];
```

Comment:
You should handle the cases when a user submits an invalid email that doesn't contain a `@`.
What happens if they submit an empty email field, or:
`alexiglesias`
`alexiglesias.finsweet.com`
Or any other variation that doesn't have a `@`?

Reply:
I see this was not tackled, right?

Reply:
yeah fixing...